### ISSUE TITLE
Cannot render multiple NativeAdView components on a single screen since 0.4.0 update

### ISSUE THREAD

**Issue by RapunzlFounder** [ammarahm-ed/react-native-admob-native-ads#142]
I am trying to implement a list with multiple NativeAdView components designed to be included in the list as the user scrolls. In previous version, 0.3.9, this was working, when the Ads were loaded imperatively.

Since the update to 0.4.0, I can only get the first NativeAdView on any given screen to render. The others, it appears as if the .loadAd() method does not fire, as there are no additional methods such as a onAdFailedToLoad or onUnifiedNativeAdLoad which occur. 

My understanding is that the component should at least receive a failed to load message? After doing some research into Google AdMob Policies, this appears to have been a Google Policy that there can only be one Ad per screen, but this is no longer the case. Curious if anyone has experienced this issue or has a suggested work around. Do each of the Ad's require a different AdUnitId?

I've implemented the NativeAdView inside of a Pure.Component and reference the component as AdView and then call this.AdView.loadAd() in componentDidMount lifecycle method, which is not how the documentation describes implementing as a functional component. Could this be causing the issue?

**Additional Info**
The NativeAdView inside of a full screen modal will successfully call .loadAd() for the first Ad on the Modal, even when called on a screen that has a NativeAdView which is populated. The 2nd and 3rd instances of the NativeAdView on that Modal similarly do not load ad data.

I plan to test if this issue persists if the NativeAdView is inside of a functional component and relies upon useEffect versus a React Pure Component using lifecycle methods.

**Comment by m15r** [comment]
Also experiencing this problem since version 0.4.0.

**Comment by fdobre** [comment]
@RapunzlFounder @m15r Have you found any workaround for this? 
Do you think any of the next possible solution would work?
- using different adUnitID for each ad fix the issue
- using React Pure Component
- downgrading to 0.39

**Comment by RapunzlFounder** [comment]
@fdobre We couldn't downgrade to 0.39 because it caused breaking changes with other project dependencies for Android Devices.

Using a React Pure Component with 0.40 works for one ad, although AdView.loadAd() does not seem to consistently fetch an ad in componentDidMount when working on a simulator.

I haven't tried a different adUnitID but I believe that will not solve the issue. The issue is related to the listener determining if the Ad is in the viewable screen for the user, and the listeners get caught on each other.

**Comment by fdobre** [comment]
I've tested using multiple adUnits when displaying multiple ads but only first ad is displayed.

**Comment by ammarahm-ed** [comment]
@RapunzlFounder have you tried the example app? It demonstrates loading ads in a list etc. Each AdViews needs to have its own ref and on each ref you need to call loadAd separately. Ads do not load if you are not calling loadAd on all AdViews rendered.

**Comment by alexmngn** [comment]
I'm facing the same issue - I've created a component that I try to use multiple times on the same screen but only the first one is rendering. 

With the following, only the first one renders:

```ts
const MyNativeAdView = () => {
  const nativeAdViewRef = useRef();

  useEffect(() => {
    nativeAdViewRef.current?.loadAd();
  }, []);

  return (
    <NativeAdView ref={nativeAdViewRef} adUnitID={adUnitId}>
      <HeadlineView style={styles.headline} />
    </NativeAdView>
  );
}
```

```ts
import MyNativeAdView from './MyNativeAdView';
//...
return (
  <>
    <MyNativeAdView />
    <MyNativeAdView />
    <MyNativeAdView />
  </>
)
```

**Comment by ammarahm-ed** [comment]
> I'm facing the same issue - I've created a component that I try to use multiple times on the same screen but only the first one is rendering.
> 
> With the following, only the first one renders:
> 
> ```ts
> const MyNativeAdView = React.memo(() => {
>   const nativeAdViewRef = useRef();
> 
>   useEffect(() => {
>     nativeAdViewRef.current?.loadAd();
>   }, []);
> 
>   return (
>     <NativeAdView ref={nativeAdViewRef} adUnitID={adUnitId}>
>       <HeadlineView style={styles.headline} />
>     </NativeAdView>
>   );
> });
> ```
> 
> ```ts
> import MyNativeAdView from './MyNativeAdView';
> //...
> return (
>   <>
>     <MyNativeAdView />
>     <MyNativeAdView />
>     <MyNativeAdView />
>   </>
> )
> ```

I need more info. 

1. Does it happen on android and ios both?
2. You should set `adUnitID` using `setRequestConfiguration`.
3. What happens it each `NativeAdView is wrapped in another `View`?. 
4. Have you seen the implementation in Example app. It loads ads in a List?

**Comment by Dheeeraj** [comment]
> 
> I need more info.
> 
> 1. Does it happen on android and ios both?
> 2. You should set `adUnitID` using `setRequestConfiguration`.
> 3. What happens it each `NativeAdView is wrapped in another `View`?.
> 4. Have you seen the implementation in Example app. It loads ads in a List?

i am using 
`` "react-native-admob-native-ads":"0.4.1"
     "react-native": "0.63.4",
    "@react-native-firebase/admob": "^11.3.2", (not sure if it matters)
``
Every 5th element loaded for me. remaining never occupied with ad

can Conform
1. Only happening in IOS, android works as expected.
2. Tried it same result.
3. i wrapped ad component with a view and also gave them unique key for both View and also the NativeAdView.
4. Yes example is working fine on its own but when i copy as it is into my project, the same thing happened.

**Comment by rogerkerse** [comment]
I have same issue, when I add multiple ads, then only few of them call `onAdLoaded` or `onAdFailedToLoad`. Most of the ads just stay quite.

**Comment by ammarahm-ed** [comment]
@rogerkerse platform?

**Comment by bartzy** [comment]
@ammarahm-ed This happens to me as well on iOS, version 0.4.1 of the library.
From 5 ads in a list, 2-3 would not load, not fire any failed/loaded callbacks.

My implementation: https://gist.github.com/bartzy/3ad636795d9f1aba43eff72ca9508ab7

**Comment by ammarahm-ed** [comment]
PRS are welcome to fix the issue.

**Comment by ammarahm-ed** [comment]
Fixed in v0.5.0
